autogpt_platform/frontend/src/components/TypeBasedInput.tsx

```diff
@@ -0,0 +1,428 @@
+import React, { FC } from "react";
+import { cn } from "@/lib/utils";
+import { format } from "date-fns";
+import { CalendarIcon } from "lucide-react";
+import { Cross2Icon, FileTextIcon } from "@radix-ui/react-icons";
+
+import { Input as DefaultInput } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import { Switch } from "@/components/ui/switch";
+import { Button } from "@/components/ui/button";
+import {
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+} from "@/components/ui/popover";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectContent,
+  SelectItem,
+} from "@/components/ui/select";
+import { determineDataType, DataType } from "@/lib/autogpt-server-api/types";
+import { BlockIOSubSchema } from "@/lib/autogpt-server-api/types";
+
+/**
+ * A generic prop structure for the TypeBasedInput.
+ *
+ * onChange expects an event-like object with e.target.value so the parent
+ * can do something like setInputValues(e.target.value).
+ */
+export interface TypeBasedInputProps {
+  schema: BlockIOSubSchema;
+  value?: any;
+  placeholder?: string;
+  onChange: (value: any) => void;
+}
+
+const inputClasses = "min-h-11 rounded-[55px] border px-4 py-2.5";
+
+const Input = function Input({
+  className,
+  ...props
+}: React.InputHTMLAttributes<HTMLInputElement>) {
+  return <DefaultInput {...props} className={cn(inputClasses, className)} />;
+};
+
+/**
+ * A generic, data-type-based input component that uses Shadcn UI.
+ * It inspects the schema via `determineDataType` and renders
+ * the correct UI component.
+ */
+export const TypeBasedInput: FC<TypeBasedInputProps> = ({
```

Comment:
Why this separate wrapper component? This can be merged with `_TypeBasedInput`

Reply:
how ?

Reply:
```tsx
switch (dataType) {
  case DataType.NUMBER:
    inputElement = (
      <Input
        type="number"
        value={value ?? ""}
        placeholder={placeholder}
        onChange={(e) => onChange(Number(e.target.value))}
      />
    );
    break;

// ...

return <div className="no-drag relative flex">{inputElement}</div>
```

Reply:
I feel like we are moving back to the squire one where we render the content using a non-component function, the only difference is we are inlining it as a local variable. Updated though,